packages/tracing/src/idletransaction.ts

```diff
@@ -220,13 +257,7 @@ export class IdleTransaction extends Transaction {
       // We need to add the timeout here to have the real endtimestamp of the transaction
       // Remember timestampWithMs is in seconds, timeout is in ms
       const end = timestampWithMs() + timeout / 1000;
-
-      setTimeout(() => {
-        if (!this._finished) {
-          this.setTag(FINISH_REASON_TAG, IDLE_TRANSACTION_FINISH_REASONS[1]);
-          this.finish(end);
-        }
-      }, timeout);
+      this._startIdleTimeout(end);
```

Comment:
```suggestion
      this._startIdleTimeout(endTimestamp);
```

(It won't let me fix the line above, because it hasn't been changed, but there, too.)